api/src/test/java/org/openmrs/module/eptsreports/reporting/intergrated/library/GenericCohortQueriesTest.java

```diff
@@ -0,0 +1,278 @@
+import org.openmrs.module.reporting.common.DateUtil;
+import org.openmrs.module.reporting.common.SetComparator;
+import org.openmrs.module.reporting.evaluation.EvaluationException;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class GenericCohortQueriesTest extends DefinitionsTest {
+  @Autowired private GenericCohortQueries genericCohortQueries;
+
+  @Autowired private TestsHelper testsHelper;
+
+  private ConceptService conceptService;
+
+  private EncounterService encounterService;
+
+  private ProgramWorkflowService programWorkflowService;
+
+  private LocationService locationService;
+
+  @Before
+  public void init() throws Exception {
+    executeLargeDataSet("genericCohortQueriesTest.xml");
+    conceptService = Context.getConceptService();
+    encounterService = Context.getEncounterService();
+    programWorkflowService = Context.getProgramWorkflowService();
+    locationService = Context.getLocationService();
+  }
+
+  @Test
+  public void hasCodedObsShouldMatchQuestion() throws EvaluationException {
+    assertEquals(
+        new HashSet<>(Arrays.asList(7)),
+        evaluateCohortDefinition(
+                genericCohortQueries.hasCodedObs(conceptService.getConcept(18), null), null)
+            .getMemberIds());
+  }
+
+  @Test
+  public void hasCodedObsShouldMatchAnswers() throws EvaluationException {
+    assertEquals(
+        new HashSet<>(Arrays.asList(7)),
+        evaluateCohortDefinition(
+                genericCohortQueries.hasCodedObs(
+                    conceptService.getConcept(18), Arrays.asList(conceptService.getConcept(7))),
+                null)
+            .getMemberIds());
+  }
+
+  @Test
+  public void hasCodedObsShouldMatchEncounterType() throws EvaluationException {
+    assertEquals(
+        new HashSet<>(Arrays.asList(7)),
+        evaluateCohortDefinition(
+                genericCohortQueries.hasCodedObs(
+                    conceptService.getConcept(18),
+                    null,
+                    null,
+                    Arrays.asList(encounterService.getEncounterType(1)),
+                    null),
+                null)
+            .getMemberIds());
+  }
+
+  @Test
+  public void hasCodedObsShouldMatchAnswersAndEncounterType() throws EvaluationException {
+    assertEquals(
+        new HashSet<>(Arrays.asList(7)),
+        evaluateCohortDefinition(
+                genericCohortQueries.hasCodedObs(
+                    conceptService.getConcept(18),
+                    null,
+                    SetComparator.IN,
+                    Arrays.asList(encounterService.getEncounterType(1)),
+                    Arrays.asList(conceptService.getConcept(7))),
+                null)
+            .getMemberIds());
+  }
+
+  @Test
+  public void createInProgramShouldReturnPatientsInAGivenProgram() throws EvaluationException {
+    Map<Parameter, Object> parameters = new HashMap<>();
```

Comment:
good idea @ynurmahomed, here is why i didn't in this case; as you down, i use similar parameters with varying values, i tend to declare variables when i want to re-use them rather than declaring everything.

Reply:
Maybe the other scenarios could be in different test cases?

Reply:
it definitely could have made more sense but with integrated tests, i try to minimise the test cases since each takes reasonable time to initiate

Reply:
Alright. I'm going to merge it.